cmd/demo/node.go

```diff
@@ -258,7 +259,23 @@ func (n *node) listen() error {
 	if err != nil {
 		return errors.WithMessage(err, "could not start tcp listener")
 	}
-	go n.client.HandleChannelProposals(n)
+
+	go n.client.Handle(n, (*UpdateHandler)(n))
+	n.client.OnNewChannel(func(ch *client.Channel) {
+		// Retrive to which of our peers this channel belongs
+		if n.pendingPeer == nil {
+			panic("Accepted channel for unknown peer")
```

Comment:
This will not work when restoring channels... also this `pendingPeer` state variable is not that nice. We probably should add a method like `Peers() []peer.Address` to the `client.Channel` type and use it in this callback.

Reply:
But i need the `perunID` here not the off-chain Addresses.

Reply:
I created a `pendingChannels map[channel.ID]*client.Channel` field now, maybe that is a bit better.